src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
As a reference, here is how NGINX processes the Host header: https://github.com/nginx/nginx/blob/46336296e4d695cf34c55ced54e21f7647a53fdc/src/http/ngx_http_request.c#L1921

Seems to me that if a `:` appears outside of square brackets, nginx will set it's state to `sw_rest` and everything before it is the host. So it doesn't seem to try to parse ipv6 addresses outside of square brackets to me. If NGINX doesn't try to correct people, I don't think we should.